### PR TITLE
85 match room chat back

### PR DESCRIPTION
**Pull request by ahmedirheem**
relates #85

### CODE REVIEW

File: server/services/matchChat.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+const getMessageByIdService = async (messageId: number) => {
+  const message = await Message.findOne({ where: { id: messageId } });
+
+  return message;
+};
+
+const getAllMessagesService = async () => {
+  const messages = await Message.findAll();
+
```
Comment (Saeed99Madi) [inline server/services/matchChat.ts:27]: You must take in consederation the not happy reults in your services, like returning no messages, in this case you need to return message means that there is no any messages! 
and return this form of response from the service : 
```
//in failure case  
return {
      status: errorStatue[ex: 404],
      data: ''no messages !",
    };
    //in success case  
return {
      status: success Status [ex:  200],
      data: result,
    };

```  
   and handle it in the controller in this way : 
   
```
 const data = (await yourService(req));

  res.status(data.status).json(data);
```
  
  this way will help you also to send the right status and clear response for the frontEnd Side
  
 ### Note : 
  - take in consideration that make this in all services and controllers

File: server/routes/matchChat.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  editMessage,
+  getAllMessages,
+  getMessageById,
+} from '../controllers';
+
+const chatRouter: Router = Router();
+
+chatRouter.get('/', getAllMessages);
+chatRouter.get('/:id', getMessageById);
+chatRouter.post('/add', addMessage);
```
Comment (Saeed99Madi) [inline server/routes/matchChat.ts:14]: `chatRouter.post('/add', addMessage);`
`add` not necessary 
for best practice to in router naming convention 
`chatRouter.post('/', addMessage);`

**Review by Saeed99Madi — CHANGES_REQUESTED** [review]

File: server/controllers/matchChat.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { Request, Response } from 'express';
+import {
+  addMessageService,
+  deleteMessageService,
+  getAllMessagesService,
+  getMessageByIdService,
+  editMessageService,
+} from '../services';
+
+const addMessage = async (req: Request, res: Response) => {
```
Comment (alaaahmade) [inline server/controllers/matchChat.ts:10]: this controller missing types and returning types :
addMessage: RequestHandler = (): Promises<Void> =>

File: server/controllers/matchChat.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  res.status(201).json({
+    status: 201,
+    data: {
+      message: 'Message added successfully',
+      newMessage,
+    },
+  });
+};
+
+const getMessageById = async (req: Request, res: Response) => {
```
Comment (alaaahmade) [inline server/controllers/matchChat.ts:28]: also this controller missing types
and all controllers on this file

File: server/controllers/matchChat.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+  res.status(200).json({
+    status: 200,
+    data: {
+      messages,
+    },
+  });
+};
+
+const deleteMessage = async (req: Request, res: Response) => {
```
Comment (alaaahmade) [inline server/controllers/matchChat.ts:52]: and this controller missing types

File: server/services/matchChat.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import { IMatchMessage } from '../interfaces';
+import { Message } from '../models';
+import { CustomError } from '../utils';
+
+const addMessageService = async ({
```
Comment (alaaahmade) [inline server/services/matchChat.ts:5]: add the returning types for all services on this file: Promise<{
"your types for the response object"
}>

**Review by alaaahmade — CHANGES_REQUESTED** [review]

**Review by Saeed99Madi — APPROVED** [review]

File: server/controllers/matchChat.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  res.status(data?.status).json(data);
+};
+
+const editMessage: RequestHandler = async (
+  req: Request,
+  res: Response,
+): Promise<void> => {
+  const { id, newMessage } = req.body;
+
+  const data = await editMessageService(id, newMessage);
```
Comment (Emanalabsi) [inline server/controllers/matchChat.ts:63]: updatedMessage would be more descriptive

**Review by Emanalabsi — APPROVED** [review]

**Review by alaaahmade — APPROVED** [review]

**Pull request merged**